stuf_zds_payments/plugin.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from typing import Any
+
+from django.template.loader import render_to_string
+from django.utils.translation import gettext_lazy as _
+
+from openforms.registrations.contrib.stuf_zds.plugin import (
+    StufZDSRegistration,
+    ZaakOptionsSerializer,
+)
+from openforms.registrations.registry import register
+from openforms.submissions.models import Submission
+from openforms.template import render_from_string
+from openforms.template.validators import DjangoTemplateValidator
+from openforms.variables.service import get_static_variables
+from rest_framework import serializers
+from stuf.stuf_zds.client import ZaakOptions
+
+from .client import get_client
+from .registration_variables import register as variables_registry
+
+logger = logging.getLogger(__name__)
+
+PLUGIN_IDENTIFIER = "stuf-zds-create-zaak:ext-utrecht"
+
+
+def get_payment_status_update_text() -> str:
+    return render_to_string(
+        "registrations/contrib/stuf_zds_payments/payment_status_update_xml.txt"
+    ).strip()
+
+
+def prepare_value(value: Any):
+    match value:
+        case bool():
+            return "true" if value else "false"
+        case list():
+            return " ".join(value)
+        case float():
+            return str(value)
+        case _:
+            return value
+
+
+class ZaakPaymentOptionsSerializer(ZaakOptionsSerializer):
+    payment_status_update_xml = serializers.CharField(
+        label=_("payment status update XML template"),
+        help_text=_(
+            "This template is evaluated with the submission data and the resulting XML "
+            "is sent to the StUF-ZDS with a PATCH to update the payment field."
+        ),
+        validators=[
+            DjangoTemplateValidator(
+                backend="openforms.template.openforms_backend",
+            ),
+        ],
+        required=False,
+    )
+
+    @classmethod
+    def display_as_jsonschema(cls):
+        data = super().display_as_jsonschema()
+        # Workaround because drf_jsonschema_serializer does not pick up defaults
+        data["properties"]["payment_status_update_xml"][
+            "default"
+        ] = get_payment_status_update_text()
+        return data
+
+
+@register(PLUGIN_IDENTIFIER)
+class StufZDSPaymentsRegistration(StufZDSRegistration):
```

Comment:
as remarked in the client, I'm afraid we also need changes to the `register_submission` call.

Perhaps a small refactor to break out a new `get_extra_data` method in Open Forms itself can make our life much simpler, I'm okay with backporting such a refactor to the 2.7.x branch. Then the only thing this subclass would have to override would probably be this new method to include the payment data.

Signature would be:

```py
def get_extra_data(self, submission: Submission) -> dict[str, Any]: ...
```

Looking at `src/stuf/stuf_zds/templates/stuf_zds/soap/includes/extraElementen.xml` and the StUF standard (https://vng-realisatie.github.io/StUF-onderlaag/documenten/Stuf0301.pdf), the values must be strings - essentially extraElementen is a key-value pair. I had first thought we could drop any XML in there, but this is possible only through `aanvullendeElementen` (and that's much more complicated, so let's ignore this).

I think this simplifies the configuration aspect in the UI - instead of giving a template, we should probably just allow them to specify the key + variable to fill out/select in a dropdown (maybe?). I think we can do this with a nested serializer, similar to the `to_emails` of the email registration plugin for the list behaviour.

Reply:
I added `get_extra_data` in https://github.com/open-formulieren/open-forms/pull/4581 and also used this in `update_payment_status`. I think this should be sufficient? It works both for updating and creating Zaken with the extra attributes

Reply:
merged and backport that PR - seems like that would indeed be the minimal approach :)